### PR TITLE
Remove liquidity callback

### PR DESCRIPTION
**Pull request by NoahZinsmeister**

### CODE REVIEW

**Comment by NoahZinsmeister** [comment]
stacked on top of #830

File: src/state/burn/hooks.ts [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+import { useTotalSupply } from '../../data/TotalSupply'
+
+const ZERO = JSBI.BigInt(0)
+
+export function useBurnState(): AppState['burn'] {
+  return useSelector<AppState, AppState['burn']>(state => state.burn)
+}
+
+export function useDerivedBurnInfo(): {
+  tokens: { [field in Exclude<Field, Field.LIQUIDITY_PERCENT | Field.LIQUIDITY>]?: Token }
```
Comment (moodysalem) [inline src/state/burn/hooks.ts:22]: Something like Pick seems more appropriate

File: src/state/burn/hooks.ts [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    independentField,
+    typedValue,
+    [Field.TOKEN_A]: { address: tokenAAddress },
+    [Field.TOKEN_B]: { address: tokenBAddress }
+  } = useBurnState()
+
+  // tokens
+  const tokenA = useTokenByAddressAndAutomaticallyAdd(tokenAAddress)
+  const tokenB = useTokenByAddressAndAutomaticallyAdd(tokenBAddress)
+  const tokens: { [field in Exclude<Field, Field.LIQUIDITY_PERCENT | Field.LIQUIDITY>]?: Token } = useMemo(
```
Comment (moodysalem) [inline src/state/burn/hooks.ts:45]: same here with pick

File: src/state/burn/hooks.ts [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+  const pair = usePair(tokens[Field.TOKEN_A], tokens[Field.TOKEN_B])
+  const noLiquidity =
+    pair === null || (!!pair && JSBI.equal(pair.reserve0.raw, ZERO) && JSBI.equal(pair.reserve1.raw, ZERO))
+
+  // route
+  const route =
+    !noLiquidity && pair && tokens[Field.TOKEN_A] ? new Route([pair], tokens[Field.TOKEN_A] as Token) : undefined
+
+  // balances
+  const relevantTokenBalances = useTokenBalancesTreatWETHAsETH(account ?? undefined, [pair?.liquidityToken])
```
Comment (moodysalem) [inline src/state/burn/hooks.ts:63]: can we do `useTokenBalance`?

**Pull request merged**